packages/eddsa-proof/src/get-snark-artifacts.node.ts

```diff
@@ -0,0 +1,22 @@
+/* istanbul ignore file */
+import download from "download"
+import tmp from "tmp"
+import fs from "fs"
+import { defaultSnarkArtifacts } from "./config"
+
+export default async function getSnarkArtifacts() {
+    const tmpDir = "eddsa-proof"
+    const tmpPath = `${tmp.tmpdir}/${tmpDir}`
+
+    if (!fs.existsSync(tmpPath)) {
```

Comment:
It misses some checks that were recently added here:

https://github.com/privacy-scaling-explorations/zk-kit/blob/a30fa3b65f3508df6a84fd6b8444f98a054e5977/packages/poseidon-proof/src/get-snark-artifacts.node.ts#L11-L24

Reply:
1- Maybe this part is not necessary because the `download` package should create the folder:
https://github.com/privacy-scaling-explorations/zk-kit/blob/a30fa3b65f3508df6a84fd6b8444f98a054e5977/packages/poseidon-proof/src/get-snark-artifacts.node.ts#L11-L13

2- Regarding the other check, the issue I see is that if someone updates the package in an existing project and the artifacts are different, then the project will not get the new artifacts:
https://github.com/privacy-scaling-explorations/zk-kit/blob/a30fa3b65f3508df6a84fd6b8444f98a054e5977/packages/poseidon-proof/src/get-snark-artifacts.node.ts#L15

Let me know what you think.

Reply:
1. Umh could you check?

2. That can only happen if a dev who is working on zk-kit updates the circuit right? I think it'll happen rarely, and in those cases, they could remove that folder manually. Otherwise, the only solution for this would be overwriting those artifacts every time a proof is generated, which makes it much less fast.

Reply:
1- I double checked and yes, the folder is created if it does not exist.

2- Yes, exactly, that will only happen if the artifacts change. I think we can add that check and keep this issue in mind.

Reply:
Oh ok 👍🏽

Reply:
We only needed to check the existence of the folder because `download` can still create the folder, then we only need this: 

```ts
if (!fs.existsSync(tmpPath) || fs.readdirSync(tmpPath).length !== 2) {
```